src/server/supplements/har/harTracer.ts

```diff
@@ -212,6 +212,15 @@ export class HarTracer {
       receive,
     };
     harEntry.time = [dns, connect, ssl, wait, receive].reduce((pre, cur) => cur > 0 ? cur + pre : pre, 0);
+
+    const promise = response.finished().then(() => {
+      const serverIPAddressAndPort = response.serverIPAddressAndPort();
+      harEntry.serverIPAddress = serverIPAddressAndPort.ipAddress === '' ? undefined : serverIPAddressAndPort.ipAddress;
+      harEntry.connection = serverIPAddressAndPort.port === -1 ? undefined : serverIPAddressAndPort.port.toString();
+      harEntry.securityDetails = response.securityDetails();
+    }).catch(() => {});
+    this._addBarrier(page, promise);
+
```

Comment:
WK only gives us the IP info later in the request lifecycle. Should we change `response.serverIPAddressAndPort()` and `response.securityDetails()` to return promises that internally wait on finished? Without that, if you call those methods for WK without waiting for finished(), you'll get the placeholder values.